src/ShowComand.php

```diff
@@ -49,27 +54,28 @@ protected function getMovieInfo(string $url, OutputInterface $out)
         $status = $response->getStatusCode();
 
         if ($status !== 200) {
-            $out->writeln(
-                "Oops! Looks like something went wrong ({$status})"
-            );
-
-            exit(EXIT_FAILURE);
+            return [null, $status];
         }
 
-        return json_decode($response->getBody());
+        return [new Movie($response), $status];
```

Comment:
I would rather stick to return the Movie or null. Think about isolate things in one method. If you return the Movie or null, you are encapsulating all the logic about API call to fetch movie inside this method and you are not leaking information (returnting the status code) about how it's implemented to the users of this method